sdk/core/azure-core/src/main/java/com/azure/core/util/implementation/ReferenceManagerImpl.java

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.implementation;
+
+import com.azure.core.util.ReferenceManager;
+
+import java.lang.ref.PhantomReference;
+import java.lang.ref.ReferenceQueue;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Implementation of {@link ReferenceManager}.
```

Comment:
It would pay to note in the docs here that this is for Java 8 and lower, and there exists one for 9+ in the appropriate MR-JAR location (otherwise I can see frustrated devs wondering why this code isn't running!)